Intervalos/src/LM/intervalos.jl

```diff
@@ -0,0 +1,248 @@
+#Definimos la estructura Intervalo
+
+struct Intervalo{T<:Real} 
+    #Definimos los espacios para el infimo y el supremo.
+    infimo::T
+    supremo::T
+    
+    function Intervalo(a, b)# where {T<:Real, S<:Real}
+#        infimo, supremo = valida_intervalo(a, b)
+        aa, bb, _ = promote(a, b, 1.0)
+        bb < aa && error("El supremo tiene que ser menor al infimo")
+        new{typeof(aa)}(aa, bb)
+    end
+end
+#Creamos una función para los intervalos delgados
+Intervalo(x) = Intervalo(x,x)
+
+intervalo_vacio(a::Intervalo)=intervalo_vacio(typeof(getfield(a, :infimo)))
+
+export Intervalo
+
+#Creamos la función intervalo_vacio
+function intervalo_vacio(x)
```

Comment:
```suggestion
function intervalo_vacio(x::Type)
```
Aqui debes decirle a la función el método a considerar, en este caso le debes decir que es una función para un tipo (ejemplo Float, Real.. etc) es por esto que debes definir cada método hay test que ya pasarían pero estan apuntando a ottra función.